docs/guides/go-built-in.md

```diff
@@ -465,14 +447,14 @@ included in blocks, it groups some of these transactions and then gives the appl
 to modify the group by invoking `PrepareProposal`.
 
 The application is free to modify the group before returning from the call, as long as the resulting set
-does not use more bytes than `RequestPrepareProposal.max_tx_bytes'
+does not use more bytes than `RequestPrepareProposal.max_tx_bytes`
 For example, the application may reorder, add, or even remove transactions from the group to improve the
 execution of the block once accepted.
 In the following code, the application simply returns the unmodified group of transactions:
 
 ```go
-func (app *KVStoreApplication) PrepareProposal(proposal abcitypes.RequestPrepareProposal) abcitypes.ResponsePrepareProposal {
-	return abcitypes.ResponsePrepareProposal{Txs: proposal.Txs}
+func (app *KVStoreApplication) PrepareProposal(_ context.Context, proposal *abcitypes.RequestPrepareProposal) (*abcitypes.ResponsePrepareProposal, error) {
+    return &abcitypes.ResponsePrepareProposal{Txs: proposal.Txs}, nil
```

Comment:
There is a problem here, which is also present before this PR.
The spec requires the App to make sure the size of `Txs` does not exceed `RequestPrepareProposal.MaxTxByte`. So a correct implementation should go through the received transactions and exclude those that don't fit.
For an example on how we do that, check [this](https://github.com/cometbft/cometbft/blob/v0.37.x/test/e2e/app/app.go#L273-L290)

Reply:
You may want to tackle this comment in a different PR, so it can be backported to both `v0.38.x` and `v0.37.x`